core/trino-main/src/main/java/io/trino/server/protocol/QueryResultRows.java

```diff
@@ -375,9 +374,9 @@ private Object getLegacyValue(Object value, Type type)
                     return value;
                 }
 
-                return unmodifiableList(((List<Object>) value).stream()
+                return ((List<Object>) value).stream()
                         .map(element -> getLegacyValue(element, elementType))
-                        .collect(toList()));
+                        .collect(toImmutableList());
```

Comment:
Careful with this one, immutable lists don’t allow null values

Reply:
yep, hence: https://github.com/trinodb/trino/pull/10680#discussion_r787923813

Reply:
any null-friendliness-driven `toList()` usage should be documented in the code.

Reply:
Yeah, this one turns out to require null-friendliness. Fortunately, it's covered by unit tests.

Reply:
I misunderstood that referenced comment, sorry :)